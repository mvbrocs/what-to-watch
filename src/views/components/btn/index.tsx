import React, { ReactNode } from 'react';

type BtnProps = {
  children: ReactNode;
  onClick: () => void;
};

const Btn = ({ children, onClick }: BtnProps) => {
  return (
    <button className="btn movie-card__button" type="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Btn;
