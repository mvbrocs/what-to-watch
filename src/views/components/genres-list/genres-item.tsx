import React, { MouseEventHandler } from 'react';
import classNames from 'classnames';

type GenresItemProps = {
  active: boolean;
  text: string;
  onClick: MouseEventHandler<HTMLAnchorElement>;
};

export const GenresItem = ({ active, text, onClick }: GenresItemProps) => (
  <li
    className={classNames('catalog__genres-item', {
      'catalog__genres-item--active': active,
    })}
  >
    <a href="/" className="catalog__genres-link" onClick={onClick}>
      {text}
    </a>
  </li>
);
