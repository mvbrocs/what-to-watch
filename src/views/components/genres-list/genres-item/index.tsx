export { GenresItem } from './genres-item';
