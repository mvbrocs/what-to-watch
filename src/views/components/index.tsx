export { default as GenresList } from './genres-list';
export { default as Header } from './header';
export { default as Logo } from './logo';
export { default as MovieCard } from './movie-card';
export { default as MovieList } from './movie-list';
export { default as Tabs } from './tabs';
export { default as TabPane } from './tabs/tab-pane';
export { default as UserBlock } from './user-block';
export { default as VideoPlayer } from './video-player';
export { default as VideoPlayerFullscreen } from './video-player-fullscreen';
export { default as Btn } from './btn';
