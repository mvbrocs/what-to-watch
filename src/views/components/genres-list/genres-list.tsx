import React from 'react';

import { GenresItem } from './genres-item';
import { GenreClickHandler } from './';

type GenresListProps = {
  genres: string[];
  activeGenre: string;
  onGenreClick: GenreClickHandler;
};

export const GenresList = ({ genres, activeGenre, onGenreClick }: GenresListProps) => (
  <ul className="catalog__genres-list">
    {genres.map((genre, index) => (
      <GenresItem
        key={index}
        active={activeGenre === genre}
        text={genre}
        onClick={onGenreClick(genre)}
      />
    ))}
  </ul>
);
