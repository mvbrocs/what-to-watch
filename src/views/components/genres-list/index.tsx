import React, { MouseEventHandler } from 'react';

import { GenresItem } from './genres-item';

export type GenreClickHandler = (genre: string) => MouseEventHandler<HTMLAnchorElement>;

type GenresListProps = {
  genres: string[];
  activeGenre: string;
  onGenreClick: GenreClickHandler;
};

const GenresList = ({ genres, activeGenre, onGenreClick }: GenresListProps) => (
  <ul className="catalog__genres-list">
    {genres.map((genre, index) => (
      <GenresItem
        key={index}
        active={activeGenre === genre}
        text={genre}
        onClick={onGenreClick(genre)}
      />
    ))}
  </ul>
);

export { GenresList };
