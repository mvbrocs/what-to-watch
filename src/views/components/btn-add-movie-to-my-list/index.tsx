import React from 'react';

import { Btn } from 'src/views/components';
import { AddIcon, InListIcon } from 'src/views/icons';
import { IFilm } from 'src/api/films';
import { useBtnAddMovieToMyList } from './use-btn-add-movie-to-my-list';

type BtnAddMovieToMyListProps = {
  movie: IFilm | null;
};

const BtnAddMovieToMyList = ({ movie }: BtnAddMovieToMyListProps) => {
  const { movieInList, clickHandler } = useBtnAddMovieToMyList(movie);
  const btnAddIcon = movieInList ? <InListIcon /> : <AddIcon />;

  return (
    <Btn onClick={clickHandler}>
      {btnAddIcon}
      <span>My list</span>
    </Btn>
  );
};

export default BtnAddMovieToMyList;
