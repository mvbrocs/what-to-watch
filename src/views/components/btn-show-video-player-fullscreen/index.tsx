import React from 'react';

import { Btn } from 'src/views/components/index';
import { PlaySIcon } from 'src/views/icons';
import { useBtnShowVideoPlayerFullscreen } from './use-btn-show-video-player-fullscreen';
import { IFilm } from 'src/api/films';

type BtnShowVideoPlayerFullscreenProps = {
  movie: IFilm | null;
};

const BtnShowVideoPlayerFullscreen = ({ movie }: BtnShowVideoPlayerFullscreenProps) => {
  const { clickHandler } = useBtnShowVideoPlayerFullscreen(movie);

  return (
    <Btn onClick={clickHandler}>
      <PlaySIcon />
      <span>Play</span>
    </Btn>
  );
};

export default BtnShowVideoPlayerFullscreen;
