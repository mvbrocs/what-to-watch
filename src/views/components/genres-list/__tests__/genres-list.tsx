import React from 'react';
import { mount } from 'enzyme';

import { GenresList } from 'views/components/genres-list';

describe('GenresList', () => {
  const mockOnGenreClick = jest.fn();
  const props = {
    genres: ['Genre 1', 'Genre 2', 'Genre 3', 'Genre 4'],
    activeGenre: 'Genre 2',
    onGenreClick: () => mockOnGenreClick,
  };
  const genresList = mount(<GenresList {...props} />);

  it('renders properly', () => {
    expect(genresList).toMatchSnapshot();
  });

  it('should render active class on item 2', () => {
    expect(
      genresList
        .find('.catalog__genres-item')
        .at(1)
        .hasClass('catalog__genres-item--active'),
    ).toBeTruthy();
  });

  it('should call onClick handler on link click', () => {
    const fourthItem = genresList.find('.catalog__genres-item').at(3);

    fourthItem.find('.catalog__genres-link').simulate('click', {
      preventDefault: () => {},
    });

    expect(mockOnGenreClick).toBeCalledTimes(1);
  });
});
