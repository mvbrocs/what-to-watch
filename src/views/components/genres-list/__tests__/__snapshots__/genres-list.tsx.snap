// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList renders properly 1`] = `
<GenresList
  activeGenre="Genre 2"
  genres={
    Array [
      "Genre 1",
      "Genre 2",
      "Genre 3",
      "Genre 4",
    ]
  }
  onGenreClick={[Function]}
>
  <ul
    className="catalog__genres-list"
  >
    <GenresItem
      active={false}
      key="0"
      onClick={[MockFunction]}
      text="Genre 1"
    >
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="/"
          onClick={[MockFunction]}
        >
          Genre 1
        </a>
      </li>
    </GenresItem>
    <GenresItem
      active={true}
      key="1"
      onClick={[MockFunction]}
      text="Genre 2"
    >
      <li
        className="catalog__genres-item catalog__genres-item--active"
      >
        <a
          className="catalog__genres-link"
          href="/"
          onClick={[MockFunction]}
        >
          Genre 2
        </a>
      </li>
    </GenresItem>
    <GenresItem
      active={false}
      key="2"
      onClick={[MockFunction]}
      text="Genre 3"
    >
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="/"
          onClick={[MockFunction]}
        >
          Genre 3
        </a>
      </li>
    </GenresItem>
    <GenresItem
      active={false}
      key="3"
      onClick={[MockFunction]}
      text="Genre 4"
    >
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="/"
          onClick={[MockFunction]}
        >
          Genre 4
        </a>
      </li>
    </GenresItem>
  </ul>
</GenresList>
`;
